Tests/TestConcurrencyHelpers.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+
+@Fixture struct TestConcurrencyHelpers {
+    func warnIfNotSendable<each T>(function: String = #function, _: repeat each T) {
+        print("At \(function), the captured arguments are not Sendable, it is not concurrency-safe.")
+    }
+
+    func warnIfNotSendable<each T: Sendable>(function: String = #function, _: repeat each T) {
+    }
+
+    /// Will be replaced to `Synchronization.Mutex` in future.
+    final class MockoloMutex<Value>: @unchecked Sendable {
+        private let lock = NSLock()
+        private var value: Value
+        init(_ initialValue: Value) {
+            self.value = initialValue
+        }
+    #if compiler(>=6.0)
+        borrowing func withLock<Result, E: Error>(_ body: (inout sending Value) throws(E) -> Result) throws(E) -> sending Result {
+            lock.lock()
+            defer { lock.unlock() }
+            return try body(&value)
+        }
+    #else
+        func withLock<Result>(_ body: (inout Value) throws -> Result) rethrows -> Result {
+            lock.lock()
+            defer { lock.unlock() }
+            return try body(&value)
+        }
+    #endif
+    }
+
+    struct MockoloUnsafeTransfer<Value>: @unchecked Sendable {
+        var value: Value
+        init(_ value: Value) {
+            self.value = value
+        }
+    }
+
+    struct MockoloHandlerState<Arg, Handler> {
+        var argValues: [MockoloUnsafeTransfer<Arg>] = []
+        var handler: Handler? = nil
+        var callCount: Int = 0
+    }
+}
+
+func warnIfNotSendable<each T>(function: String = #function, _: repeat each T) {
+    print("At \(function), the captured arguments are not Sendable, it is not concurrency-safe.")
+}
+func warnIfNotSendable<each T: Sendable>(function: String = #function, _: repeat each T) {
+}
+typealias MockoloMutex = TestConcurrencyHelpers.MockoloMutex
+typealias MockoloUnsafeTransfer = TestConcurrencyHelpers.MockoloUnsafeTransfer
+typealias MockoloHandlerState = TestConcurrencyHelpers.MockoloHandlerState
+
+import XCTest
+
+final class TestConcurrencyHelpersTests: MockoloTestCase {
```

Comment:
```suggestion
final class ConcurrencyHelpersTests: MockoloTestCase {
```

no hurry and nits.
In my opinion, a system in this test case is not TestConcurrencyHelper but [ConcurrencyHelper](https://github.com/sidepelican/mockolo/blob/2258ee1b13347eaf56755451703fcb0b9bc8f085/Sources/MockoloFramework/Templates/ConcurrencyHelpersTemplate.swift#L1). so this class might be `ConcurrencyHelpersTests` if I understand correctly.